daemon/SecretsImpl/secrets.cpp

```diff
@@ -1814,14 +1822,60 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
                 qCWarning(lcSailfishSecretsDaemon) << "GetStandaloneSecretRequest:" << request->requestId << "finished as pending!";
                 *completed = true;
             } else {
-                QByteArray secret = request->outParams.size()
-                        ? request->outParams.takeFirst().toByteArray()
-                        : QByteArray();
+                Secret secret = request->outParams.size()
+                        ? request->outParams.takeFirst().value<Secret>()
+                        : Secret();
                 if (request->isSecretsCryptoRequest) {
-                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<QByteArray>(secret));
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << QVariant::fromValue<Secret>(secret));
                 } else {
                     request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
-                                                                            << QVariant::fromValue<QByteArray>(secret));
+                                                                            << QVariant::fromValue<Secret>(secret));
+                }
+                *completed = true;
+            }
+            break;
+        }
+        case FindCollectionSecretsRequest: {
+            Result result = request->outParams.size()
+                    ? request->outParams.takeFirst().value<Result>()
+                    : Result(Result::UnknownError,
+                             QLatin1String("Unable to determine result of FindCollectionSecretsRequest request"));
+            if (result.code() == Result::Pending) {
+                // shouldn't happen!
+                qCWarning(lcSailfishSecretsDaemon) << "FindCollectionSecretsRequest:" << request->requestId << "finished as pending!";
+                *completed = true;
+            } else {
+                QVector<Secret::Identifier> identifiers = request->outParams.size()
+                        ? request->outParams.takeFirst().value<QVector<Secret::Identifier> >()
```

Comment:
Since C++11, the compiler can distinguish between the `>>` operator and template syntax, so you don't need to use `> >` here. Suggestion: `.value<QVector<Secret::Identifier>>()` (without the extra space between the two `>`s)

Reply:
Same goes for every similar case.

Reply:
GCC below certain versions cannot.  Slava complains when we use ">>" instead of "> >" because he tests with old compilers which don't support the syntax.  There is zero cost to using the old syntax which is supported everywhere, so I will continue to use that one.

Reply:
Ok.
I guess the question of how old compilers doesn't belong here, so let's talk about it somewhere else.